hyper/http20/connection.py

```diff
@@ -260,19 +347,18 @@ def connect(self):
 
             self._send_preamble()
 
-        return
-
     def _send_preamble(self):
         """
         Sends the necessary HTTP/2 preamble.
         """
         # We need to send the connection header immediately on this
         # connection, followed by an initial settings frame.
-        self._conn.initiate_connection()
-        self._conn.update_settings(
-            {h2.settings.ENABLE_PUSH: int(self._enable_push)}
-        )
-        self._sock.sendall(self._conn.data_to_send())
+        with self._conn as conn:
+            conn.initiate_connection()
+            conn.update_settings(
+                {h2.settings.ENABLE_PUSH: int(self._enable_push)}
+            )
+        self._send_outstanding_data()
```

Comment:
I think this call may need to be inside the `with` statement here, because `_send_outstanding_data` implicitly uses the connection object.